### PR TITLE
Correctly check the mounted pages in the `hasAccess()` method

### LINKED ISSUE #5707: Breadcrumbs in der Seitenstruktur (Backend) werden nicht korrekt verlinkt.
### Affected version(s)

4.13.8

### Description

Die Pagemounts für Benutzergruppen besagen: "Unterseiten werden automatisch hinzugefügt".

Im Seitenbaum funktioniert das auch komplett richtig. Wenn die Benutzergruppe eine Seite gemounted bekommen hat, gilt das auch für alle folgenden Unterseiten.

In den Breadcrumbs der Seitenstruktur sind aber nur die Seiten verlinkt, die der Benutzergruppe explizit zugewiesen wurden. Hier wird die Vererbung nicht berücksichtigt.

Erwartungskonform wäre, dass auch hier die Unterseiten einer gemounteten Seite verlinkt sind.

### PR DESCRIPTION
**Pull request by leofeyer**
Fixes #5707

The previous discussion is here: #2895

### CODE REVIEW

**Comment by aschempp** [comment]
what happened to my comment? https://github.com/contao/contao/issues/2895#issuecomment-1038744528

**Comment by leofeyer** [comment]
I would like the voter behave the same as the `hasAccess()` method, but we have to discuss the whole issue again anyway.

**Comment by leofeyer** [comment]
As discussed in the call, we want to merge this PR now. 😄

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Resources/contao/classes/BackendUser.php [hunk 1]
```diff
@@ -275,6 +275,19 @@ public function hasAccess($field, $array)
...
+			// Check the mounted pages
+			foreach ($this->pagemounts as $page)
+			{
+				$childIds = $this->Database->getChildRecords($page, 'tl_page');
+
+				if (!empty($childIds) && array_intersect($field, $childIds))
+				{
+					return true;
+				}
+			}
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/classes/BackendUser.php]:
```suggestion
			$childIds = $this->Database->getChildRecords($this->pagemounts, 'tl_page');

			if (!empty($childIds) && array_intersect($field, $childIds))
			{
				return true;
			}
```

**Review by aschempp — CHANGES_REQUESTED** [review]
The `getChildRecords` accepts a list of parent page IDs, so one DB request to fetch the children should be quicker than iterate.

**Review by aschempp — APPROVED** [review]

**Review by Toflar — APPROVED** [review]

**Pull request merged**